### PR TITLE
scope: parameter does not escape through itself

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

**Review by PetarKirov — APPROVED** [review]
Looks good

File: test/fail_compilation/retscope.d [hunk 1]
```diff
@@ -626,3 +626,15 @@ struct Result(R)
     n.empty();
 }
 
+/**********************************************/
```
Comment (mathias-lang-sociomantic) [inline test/fail_compilation/retscope.d]: What's the error message ?

Comment (WalterBright) [inline test/fail_compilation/retscope.d]: there isn't one. It's not supposed to generate one.

Comment (Geod24) [inline test/fail_compilation/retscope.d]: Why put in in fail_compilation then ?

Comment (WalterBright) [inline test/fail_compilation/retscope.d]: I like to group related tests into one file.

Comment (mathias-lang-sociomantic) [inline test/fail_compilation/retscope.d]: Yeah then put it in `test/runnable/testscope.d`, which is for test that are passing.

It's weird to have a test that is supposed to compile in a file that test things that should be rejected.

**Comment by andralex** [comment]
Auto-merge toggled on

**Pull request merged**